src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -78,6 +164,56 @@ public static LogicalLocationVersionOne CreateLogicalLocationVersionOne(LogicalL
             return logicalLocation;
         }
 
+        public static RuleVersionOne CreateRule(Rule v2Rule)
+        {
+            RuleVersionOne rule = null;
+
+            if (v2Rule != null)
+            {
+                rule = new RuleVersionOne
+                {
+                    Id = v2Rule.Id,
+                    MessageFormats = v2Rule.MessageStrings,
+                    Properties = v2Rule.Properties
+                };
+
+                if (v2Rule.Configuration != null)
+                {
+                    rule.Configuration = v2Rule.Configuration.Enabled ?
+                            RuleConfigurationVersionOne.Enabled :
+                            RuleConfigurationVersionOne.Disabled;
+                    rule.DefaultLevel = SarifTransformerUtilities.CreateResultLevelVersionOne(v2Rule.Configuration.DefaultLevel);
+                }
+
+                if (v2Rule.Name != null)
+                {
+                    rule.Name = v2Rule.Name.Text;
+                }
+
+                if (v2Rule.FullDescription != null)
+                {
+                    rule.FullDescription = v2Rule.FullDescription.Text;
```

Comment:
@michaelcfanning @EasyRhin0 The v2 `rule.FullDescription` property is a `message` object. That means in might contain both a plain text `text` property and a Markdown-based `richText` property. It might also have an `arguments` property. And it might even specify the message by means of its "resource identifier" properties `messageId` and `richMessageId`.

Downgrading a v2 `message` object in a way that is round-trippable back to v1 is difficult! For now, please do this:

1. Copy `v2Rule.FullDescription.RichText` to `rule.RichDescription`. At least that's an improvement.
2. File an issue "Design and implement v2 -> v1 -> v2 message round-tripping".
3. Create a GitHub issue label `converter` and label this issue with it.

... because I think we're going to find many issues like this, where we need to make progress on the easy cases without losing sight of the corner cases we'll have to deal with.

Reply:
Doesn't seem to be a rule.RichDescription property.

Reply:
You're right. That was crazy talk. I stand by #&#xfeff;2 and #&#xfeff;3, though :-)